commands/operator-sdk/cmd/new.go

```diff
@@ -80,59 +83,161 @@ const (
 	if len(projectName) == 0 {
-		cmdError.ExitWithError(cmdError.ExitBadArgs, fmt.Errorf("project-name must not be empty"))
+		log.Fatal("project-name must not be empty")
 	}
 }
 
 // mustBeNewProject checks if the given project exists under the current diretory.
 // it exits with error when the project exists.
 func mustBeNewProject() {
-	fp := filepath.Join(mustGetwd(), projectName)
+	fp := filepath.Join(cmdutil.MustGetwd(), projectName)
 	stat, err := os.Stat(fp)
 	if err != nil && os.IsNotExist(err) {
 		return
 	}
 	if err != nil {
-		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to determine if project (%v) exists", projectName))
+		log.Fatalf("failed to determine if project (%v) exists", projectName)
 	}
 	if stat.IsDir() {
-		cmdError.ExitWithError(cmdError.ExitBadArgs, fmt.Errorf("project (%v) exists. please use a different project name or delete the existing one", projectName))
+		log.Fatalf("project (%v) exists. please use a different project name or delete the existing one", projectName)
+	}
+}
+
+func doScaffold() {
+	cfg := &input.Config{
+		Repo:           filepath.Join(cmdutil.CheckAndGetCurrPkg(), projectName),
+		AbsProjectPath: filepath.Join(cmdutil.MustGetwd(), projectName),
+		ProjectName:    projectName,
+	}
+
+	s := &scaffold.Scaffold{}
+	err := s.Execute(cfg,
+		&scaffold.Cmd{},
+		&scaffold.Dockerfile{},
+		&scaffold.Role{},
+		&scaffold.RoleBinding{},
+		&scaffold.Operator{},
+		&scaffold.Apis{},
+		&scaffold.Controller{},
+		&scaffold.Version{},
+		&scaffold.Gitignore{},
+		&scaffold.GopkgToml{},
+	)
+	if err != nil {
+		log.Fatalf("new scaffold failed: (%v)", err)
+	}
+}
+
+func doAnsibleScaffold() {
+	cfg := &input.Config{
+		AbsProjectPath: filepath.Join(cmdutil.MustGetwd(), projectName),
+		ProjectName:    projectName,
+	}
+
+	resource, err := scaffold.NewResource(apiVersion, kind)
+	if err != nil {
+		log.Fatal("Invalid apiVersion and kind.")
+	}
+
+	s := &scaffold.Scaffold{}
+	tmpdir, err := ioutil.TempDir("", "osdk")
+	if err != nil {
+		log.Fatal("unable to get temp directory")
+	}
+
+	galaxyInit := &ansible.GalaxyInit{
+		Resource: *resource,
+		Dir:      tmpdir,
+	}
+
+	err = s.Execute(cfg,
+		&ansible.Dockerfile{
+			GeneratePlaybook: generatePlaybook,
+		},
+		&ansible.WatchesYAML{
+			Resource:         *resource,
+			GeneratePlaybook: generatePlaybook,
+		},
+		galaxyInit,
+		&scaffold.Role{},
+		&scaffold.RoleBinding{},
+		&ansible.Operator{},
+		&scaffold.Crd{
+			Resource: resource,
+		},
+		&scaffold.Cr{
+			Resource: resource,
+		},
+	)
+	if err != nil {
+		log.Fatalf("new scaffold failed: (%v)", err)
+	}
+
+	// Decide on playbook.
+	if generatePlaybook {
+		err := s.Execute(cfg,
+			&ansible.Playbook{
+				Resource: *resource,
+			},
+		)
+		if err != nil {
+			log.Fatalf("new scaffold failed: (%v)", err)
+		}
+	}
+
+	// Run galaxy init.
+	cmd := exec.Command(filepath.Join(galaxyInit.AbsProjectPath, galaxyInit.Path))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Run()
```

Comment:
Check return error and exit?

Or do we still need to remove the galaxy init script below?

Reply:
I think we need to remove the galaxy init script. If the command errors the right info will already be printed to the console.